packages/core-data/src/selectors.js

```diff
@@ -118,13 +118,190 @@ export function getEntityRecord( state, kind, name, key ) {
+		const edits =
+							getEntityRecordEdits( state, kind, name, recordId ) || [];
+		return Object.keys( edits ).reduce( ( acc, key ) => {
+			if ( ! transientEdits[ key ] ) {
+				acc[ key ] = edits[ key ];
+			}
+			return acc;
+		}, {} );
+	},
+	( state ) => [ state.entities.config, state.entities.data ]
+);
+
+/**
+ * Returns true if the specified entity record has edits,
+ * and false otherwise.
+ *
+ * @param {Object} state    State tree.
+ * @param {string} kind     Entity kind.
+ * @param {string} name     Entity name.
+ * @param {number} recordId Record ID.
+ *
+ * @return {boolean} Whether the entity record has edits or not.
+ */
+export function hasEditsForEntityRecord( state, kind, name, recordId ) {
+	return Object.keys( getEntityRecordNonTransientEdits( state, kind, name, recordId ) ).length > 0;
+}
+
+/**
+ * Returns the specified entity record, merged with its edits.
+ *
+ * @param {Object} state    State tree.
+ * @param {string} kind     Entity kind.
+ * @param {string} name     Entity name.
+ * @param {number} recordId Record ID.
+ *
+ * @return {Object?} The entity record, merged with its edits.
+ */
+export const getEditedEntityRecord = createSelector(
+	( state, kind, name, recordId ) => {
+		const record = getEntityRecord( state, kind, name, recordId );
+		return {
+			...Object.keys( record ).reduce( ( acc, key ) => {
+				acc[ key ] = get( record[ key ], 'raw', record[ key ] );
+				return acc;
+			}, {} ),
+			...getEntityRecordEdits( state, kind, name, recordId ),
+		};
+	},
+	( state ) => [ state.entities.data ]
+);
+
+/**
+ * Returns true if the specified entity record is saving, and false otherwise.
+ *
+ * @param {Object} state    State tree.
+ * @param {string} kind     Entity kind.
+ * @param {string} name     Entity name.
+ * @param {number} recordId Record ID.
+ *
+ * @return {Object?} Whether the entity record is saving or not.
+ */
+export function isSavingEntityRecord( state, kind, name, recordId ) {
+	return get(
+		state.entities.data,
+		[ kind, name, 'saving', recordId, 'pending' ],
+		false
+	);
+}
+
+/**
+ * Returns the specified entity record's last save error.
+ *
+ * @param {Object} state    State tree.
+ * @param {string} kind     Entity kind.
+ * @param {string} name     Entity name.
+ * @param {number} recordId Record ID.
+ *
+ * @return {Object?} The entity record's save error.
+ */
+export function getLastEntitySaveError( state, kind, name, recordId ) {
+	return get( state.entities.data, [ kind, name, 'saving', recordId, 'error' ] );
+}
+
+/**
+ * Returns the current undo offset for the
+ * entity records edits history. The offset
+ * represents how many items from the end
+ * of the history stack we are at. 0 is the
+ * last edit, -1 is the second last, and so on.
+ *
+ * @param {Object} state State tree.
+ *
+ * @return {number} The current undo offset.
+ */
+export function getCurrentUndoOffset( state ) {
+	return state.undo.offset;
```

Comment:
While I understand the purpose now, I do wonder whether this should be exposed or not, as it seems like just an implementation detail. Any particular reason for having this selector?

Reply:
To not repeat it in the following two, but yeah, it could be private for now. Although I do potentially see people using it to show how many redos are available or something like that.

Reply:
it's different than the offset though. Let's make it private for now.

Reply:
It's the negation of the offset, yeah kind of confusing. I'll make it private in the autosaves PR.